lighthouse-core/runner.js

```diff
@@ -135,27 +134,30 @@ class Runner {
    * @param {*} connection
    * @return {!Promise<!Artifacts>}
    */
-  static _gatherArtifactsFromBrowser(opts, connection) {
+  static async _gatherArtifactsFromBrowser(opts, connection) {
     if (!opts.config.passes) {
       return Promise.reject(new Error('No browser artifacts are either provided or requested.'));
     }
 
     opts.driver = opts.driverMock || new Driver(connection);
-    return GatherRunner.run(opts.config.passes, opts).then(artifacts => {
-      const flags = opts.flags;
-      const shouldSave = flags.gatherMode;
-      const p = shouldSave ? Runner._saveArtifacts(artifacts): Promise.resolve();
-      return p.then(_ => artifacts);
-    });
+    const artifacts = await GatherRunner.run(opts.config.passes, opts);
+
+    const flags = opts.flags;
+    if (flags.gatherMode) {
+      const path = Runner._getArtifactsPath(flags);
+      await Runner._saveArtifacts(artifacts, path);
```

Comment:
inherited, but this would probably make more sense back up in `run()`

Reply:
true. but how about we do that when runner goes async/await?

Reply:
Well you're already refactoring it to async/await so you brought it on yourself :P

It doesn't look like there are any tests of this method (since it's private, yay), so should pop into the surrounding else branch with no side effects?